components/cam/src/physics/cam/shoc_intr.F90

```diff
@@ -1241,6 +1277,39 @@ subroutine shoc_tend_e3sm( &
 
 #endif    
     return         
-  end subroutine shoc_tend_e3sm      
+  end subroutine shoc_tend_e3sm   
+  
+  subroutine grid_size(state, grid_dx, grid_dy)
+  ! Determine the size of the grid for each of the columns in state
+
+  use phys_grid,       only: get_area_p
+  use shr_const_mod,   only: shr_const_pi
+  use physics_types,   only: physics_state
+  use ppgrid,          only: pver, pverp, pcols
+ 
+  type(physics_state), intent(in) :: state
+  real(r8), intent(out)           :: grid_dx(pcols), grid_dy(pcols)   ! E3SM grid [m]
+
+  real(r8), parameter :: earth_ellipsoid1 = 111132.92_r8 ! first coefficient, meters per degree longitude at equator
```

Comment:
Apologies for being dense, but could you elaborate a little on where these come from?

Reply:
World Geodetic System 84 (WGS84), the reference coordinate system used by GPS.  Added a slightly more verbose comment.

Reply:
Sorry my comment wasn't very clear. I'm just trying to verify these values and the equation used, and I'm having a hard time finding a reference that shows the equation you are using. My goal is not to make your code overly verbose or anything, I was just hoping to double-check your implementation.

Reply:
See https://gis.stackexchange.com/questions/75528/understanding-terms-in-length-of-degree-formula with the accompanying references

Reply:
Okay, your implementation is consistent with that, I was just looking for a reference in the comments in the code to where one might find that equation used, because the values appear to come out of nowhere. From following the references you point to, the equation used here in the code doesn't seem to immediately fall out of WGS84, rather WGS84 defines the ellipsoid, and then some more geometry and arithmetic (which does not seem to be specific to WGS84) are needed to derive an elliptic equation that exactly determines distance, and then an expansion is used to approximate that equation, which is then truncated to third order here.